cpp/src/fil/common.cuh

```diff
@@ -165,8 +178,8 @@ struct predict_params : shmem_size_params {
   int num_outputs;
 
   // Data parameters.
-  float* preds;
-  const float* data;
+  void* preds;
```

Comment:
It's better to add a comment that both `preds` and `data` are pointers to either `float` or `double`.

Reply:
Done.